server/src/weaverbird/backends/pypika_translator/translators/postgresql.py

```diff
@@ -36,6 +38,26 @@ class PostgreSQLTranslator(SQLTranslator):
     TO_DATE_OP = ToDateOp.TO_DATE
     QUOTE_CHAR = '"'
 
+    @classmethod
+    def _add_date(cls, *, date_column: str | Field, add_date_value: int, add_date_unit: DATE_UNIT):
+        return LiteralValue(f"{date_column} + INTERVAL '{add_date_value} {add_date_unit}'")
+
+    def evolution(
+        self: Self,
+        *,
+        builder: 'QueryBuilder',
+        prev_step_name: str,
+        columns: list[str],
+        step: "EvolutionStep",
+    ) -> StepContext:
+
+        self.DATEADD_FUNC = LiteralValue(
+            f"{step.date_col} + INTERVAL '1 {self.EVOLUTION_DATE_UNIT[step.evolution_type]}'"
+        )
+        return super().evolution(  # type: ignore
+            builder=builder, prev_step_name=prev_step_name, columns=columns, step=step
+        )
```

Comment:
I don't like mutating the class in a method.. How about taking a `dateadd_func: CustomFunction | LiteralValue | None` parameter in the base translator's `evolution` function and default to `self.DATEADD_FUNC` if the parameter is None ?